io_buffer.c

```diff
@@ -2279,7 +2285,12 @@ io_buffer_get_string(int argc, VALUE *argv, VALUE self)
     rb_encoding *encoding = rb_ascii8bit_encoding();
 
     if (argc >= 1) {
-        offset = NUM2SIZET(argv[0]);
+        long offset_int = NUM2LONG(argv[0]);
+        if (offset_int < 0) {
+            offset_int = size + offset_int;
+            if (offset_int < 0) offset_int = 0;
```

Comment:
Shouldn't raise `ArgumentError`?

Reply:
As I wrote in the issue (https://bugs.ruby-lang.org/issues/19753), I think it should behave like `String#[]` or `Array#[]`, so negative offset should be acceptable, but counting from the end of the buffer.